### DISCUSSION TITLE
Potential Review Candidates.

### DISCUSSION THREAD

**Discussion by rajarshimaitra** [Bitshala/BitcoinCore-PR-Review-Club#5] [Ideas]
Please comment below PRs you want to see discussed in the club. If you have your own PR that could use some eyes, we would be happy to take a look at our club.

**Comment by stratospher** [comment]
@rajarshimaitra here's the PR notes + questions for #27374 - https://gist.github.com/stratospher/2e68d56c5005346e7fd5f5fea137e2bc

**Comment by stratospher** [comment]
# Session Details
PR #26222

# Notes
- Highly recommend reading Chapter 1 - 3 from Jimmy Song's [Programming Bitcoin](https://digilib.stekom.ac.id/assets/dokumen/ebook/feb_d82be9cf1cb52e2b294a82275318a5c8235444eb_1654093256.pdf) for understanding elliptic curve cryptography basics
  - Sections on finite field arithmetic operations, group operations, defining the curve for bitcoin in particular would be useful for reviewing this PR
- Bitcoin uses secp256k1 elliptic curve for all it's cryptographic operations like key/address generation, digital signatures etc. See [curve specifications](https://en.bitcoin.it/wiki/Secp256k1) here.
- Mathematically:
  - Elliptic curves are [groups](https://github.com/jimmysong/programmingbitcoin/blob/master/ch03.asciidoc#mathematical-groups).
  - Public keys are [group](https://cryptobook.nakov.com/asymmetric-key-ciphers/elliptic-curve-cryptography-ecc) elements. (points on the elliptic curve)
  - Points have an x coordinate and a y coordinate. Individually, each coordinate is a [field](https://github.com/jimmysong/programmingbitcoin/blob/master/ch01.asciidoc#finite-field-definition) element.
  - in the questions below, GE = group element and FE = field element
- This PR rewrites the functional test framework's elliptic curve implementation in a much more intuitive and easy to understand manner by using fields and groups.

# Questions
1. Did you review the PR? Concept ACK, approach ACK, tested ACK, or NACK? What was your review approach?
2. Write a test using `class FE` for basic field element operations - addition, multiplication.
3. Verify [Fermat's little theorem](https://github.com/jimmysong/programmingbitcoin/blob/master/ch01.asciidoc#finite-field-division) using `class FE`.
4. `class GE` represents infinity explicitly. what is point at infinity in an elliptic curve? Why do we need this point? Where are regions in the code where we need to handle infinity scenarios properly?
5. What's a generator point in an elliptic curve?
6. Are there any downsides to rewriting the elliptic curve logic using fields/groups?
